pcl_ros/include/pcl_ros/impl/transforms.hpp

```diff
@@ -92,30 +106,33 @@ template <typename PointT> bool
 transformPointCloudWithNormals (const std::string &target_frame,
                                 const pcl::PointCloud <PointT> &cloud_in,
                                 pcl::PointCloud <PointT> &cloud_out, 
-                                const tf::TransformListener &tf_listener)
+                                const tf2_ros::Buffer &tf_buffer)
 {
   if (cloud_in.header.frame_id == target_frame)
   {
     cloud_out = cloud_in;
     return (true);
   }
 
-  tf::StampedTransform transform;
+  geometry_msgs::msg::TransformStamped transform_msg;
   try
   {
-    tf_listener.lookupTransform (target_frame, cloud_in.header.frame_id, fromPCL(cloud_in.header).stamp, transform);
+    transform_msg = tf_buffer.lookupTransform (target_frame, cloud_in.header.frame_id, fromPCL(cloud_in.header.stamp));
   }
-  catch (tf::LookupException &e)
+  catch (tf2::LookupException &e)
   {
-    ROS_ERROR ("%s", e.what ());
+    RCLCPP_ERROR (rclcpp::get_logger("pcl_ros"), "%s", e.what ());
     return (false);
   }
-  catch (tf::ExtrapolationException &e)
+  catch (tf2::ExtrapolationException &e)
   {
-    ROS_ERROR ("%s", e.what ());
+    RCLCPP_ERROR (rclcpp::get_logger("pcl_ros"), "%s", e.what ());
     return (false);
   }
 
+  tf2::Transform transform;
+  tf2::convert (transform_msg.transform, transform);
+
```

Comment:
This conversion step is a candidate for a helper function, but some feedback is needed.

This is more about the tf->tf2 port than the ROS2 port. Given that the `lookupTransform` API in tf2 returns `geometry_msgs::msg::TransformStamped` instead of `tf2::Transform` directly, I wonder if this calls for the creation of a new overload of the `transformPointCloud[WithNormals]` APIs like so:

```
  template <typename PointT> void 
  transformPointCloudWithNormals (const pcl::PointCloud <PointT> &cloud_in, 
                                  pcl::PointCloud <PointT> &cloud_out, 
                                  const geometry_msgs::msg::Transform &transform);
```

Which would provide this conversion. Obviously this pulls `geometry_msgs` into the public interface.

Additionally, would the existing API which takes a `tf2::Transform` be considered deprecated?

Reply:
Sure, makes sense to support both.